packages/frontend-api/src/Model/Mutation/Customer/DeliveryAddress/DeliveryAddressMutation.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Mutation\Customer\DeliveryAddress;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressDataFactory;
+use Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressFacade;
+use Shopsys\FrameworkBundle\Model\Customer\Exception\DeliveryAddressNotFoundException;
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserFacade;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserUpdateDataFactory;
+use Shopsys\FrontendApiBundle\Model\Mutation\BaseTokenMutation;
+use Shopsys\FrontendApiBundle\Model\Mutation\Customer\DeliveryAddress\Exception\DeliveryAddressNotFoundUserError;
+use Shopsys\FrontendApiBundle\Model\Mutation\Customer\User\Exception\InvalidCredentialsUserError;
+
+class DeliveryAddressMutation extends BaseTokenMutation
```

Comment:
Thx for that, I am questioning myself now, whether I used it correctly (or as I intended). Delivery address mutations are available only for logged user. 
Shouldn't BaseTokenMutation cover scenarios, that check logged user ? Shouldn't I run 
 `runCheckUserIsLogged` in every mutation. What is a purpose of  `BaseTokenMutation` ?

Reply:
I believe you are right, I would say it should have been already used in project-base (however, it extended `AbstractMutation` there). Extending `BaseTokenMutation` seems more appropriate because of the reasons you mentioned, let's use that one :ok_hand: You will need to slightly alter the logic though because `runCheckUserIsLogged` returns an instance of `FrontendApiUser` that provides UUID, and then you can use that UUID to get an instance of `CustomerUser`.

And one more thing - you need to alter the constructor if you use  `BaseTokenMutation` (parent constructor with `TokenStorageInterface` instance should be called). Btw, I recommend using [EA inspections](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-) plugin :wink: 
![image](https://github.com/user-attachments/assets/4ad56bb4-8f66-4231-a811-1d2b11c91a3e)

Reply:
one more thought - maybe I would use `InvalidCredentialsUserError('You need to be logged in.')` inside `runCheckUserIsLogged()` method instead of the current `InvalidTokenUserMessageException`

Reply:
however, beware the usages on storefront, and in tests :thinking:

Reply:
at this point, it makes me think whether it is worth the changes (it will cause some BC breaks probably). But I will leave it up to you, and we can discuss this further of course :wink: